hypervisor/include/arch/x86/asm/guest/nested.h

```diff
@@ -333,12 +334,20 @@ struct acrn_vvmcs {
 	uint8_t vmcs02[PAGE_SIZE];	/* VMCS to run L2 and as Link Pointer in VMCS01 */
 	struct acrn_vmcs12 vmcs12;	/* To cache L1's VMCS12*/
 	uint64_t vmcs12_gpa;            /* The corresponding L1 GPA for this VMCS12 */
+	uint32_t ref_cnt;		/* Count of being VMPTRLDed without VMCLEARed */
 	bool host_state_dirty;		/* To indicate need to merge VMCS12 host-state fields to VMCS01 */
 	bool control_fields_dirty;	/* For all other non-host-state fields that need to be merged */
 } __aligned(PAGE_SIZE);
 
+/*
+ * MAX_ACTIVE_VVMCS_NUM could be of any arbitrary number. We chose MAX_VCPUS_PER_VM
+ * because it is large enough so that vvmcs[] is able to cache most active VMCS12s
+ * in most configurations.
+ */
+#define MAX_ACTIVE_VVMCS_NUM	MAX_VCPUS_PER_VM
```

Comment:
Don't link this to MAX_VCPUS_PER_VM. It is completely seperate.

Reply:
> Don't link this to MAX_VCPUS_PER_VM. It is completely seperate.

OK. I changed MAX_ACTIVE_VVMCS_NUM to 4.